urla958522fc1_spirosikmd_inmdsy_08_tgz-pJ8-TesterAppJ8/src/nl/rug/peerbox/logic/filesystem/FileRequestTask.java

```diff
@@ -0,0 +1,114 @@
+package nl.rug.peerbox.logic.filesystem;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketTimeoutException;
+import java.security.MessageDigest;
+import java.util.concurrent.TimeoutException;
+
+import nl.rug.peerbox.logic.Context;
+import nl.rug.peerbox.logic.Peer;
+
+import org.apache.commons.lang3.SystemUtils;
+import org.apache.log4j.Logger;
+
+public final class FileRequestTask implements Runnable {
+
+	private static final int SOCKET_TIMEOUT = 5000;
+
+	private static final Logger logger = Logger
+			.getLogger(FileRequestTask.class);
+
+	private final Peer h;
+	private final String filename;
+	private final Context ctx;
+	private final PeerboxFile file;
+
+	public FileRequestTask(PeerboxFile file, Context ctx) {
+		this.file = file;
+		this.h = file.getOwner();
+		this.filename = file.getFilename();
+		this.ctx = ctx;
+	}
+
+	File hide(File src) {
+		if (SystemUtils.IS_OS_WINDOWS) {
+			Process p;
+			try {
+				p = Runtime.getRuntime().exec("attrib +h " + src.getPath());
+				p.waitFor();
+			} catch (IOException | InterruptedException e) {
+				e.printStackTrace();
+			}
+		}
+		return src;
+	}
+
+	@Override
+	public void run() {
+		String tempFilename = "." + filename;
+		File tempFile;
+		try {
+			tempFile = File.createTempFile("peerbox_", filename);
+		} catch (IOException e1) {
+			tempFile = new File(ctx.getPathToPeerbox(), tempFilename);
+		}
+
+		try (Socket s = new Socket()) {
+			s.connect(new InetSocketAddress(h.getAddress(), h.getPort()),
+					SOCKET_TIMEOUT);
+			logger.debug("Created direct connection to "
+					+ h.getAddress().toString() + ":" + h.getPort());
+			PrintWriter put = new PrintWriter(s.getOutputStream(), true);
+			put.println(file.getUFID().getFilename());
+			byte[] mybytearray = new byte[1024];
+			InputStream is = s.getInputStream();
+
+			FileOutputStream fos = new FileOutputStream(tempFile);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fos*. The resource is referenced at lines 74, 78. The resource is closed at line 80. There are other execution paths that do not contain closure statements, for example, when InputStream.read throws an exception. Ensure *fos* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).